include/aspect/material_model/drucker_prager.h

```diff
@@ -0,0 +1,247 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__model_drucker_prager_h
+#define __aspect__model_drucker_prager_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model that consists of globally constant values for all
+     * material parameters except density and viscosity.
+     *
+     * The model is considered incompressible, following the definition
+     * described in Interface::is_compressible.
+     *
+     * The viscosity is computed according to the Drucker Prager frictional
+     * plasticity criterion based on a user-defined Internal angle of frication
+     * and cohesion. In 3D:
+     * $yield_strength = \\frac{6 cohesion cos(\\phi)}{\\sqrt(3) (3+sin(\\phi))} +\\
+     * \\frac{2 pressure sin(\\phi)}{\\sqrt(3) (3+sin(\\phi))}$.
```

Comment:
What is `phi` here? The formula contains text that isn't printed well in latex formulas. I would replace `pressure -> p`, `yield_strength -> \\sigma_y`, `cohesion -> ??`